src/pages/index.js

```diff
@@ -1,93 +1,236 @@
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 
 //define global variables
 const profileModal = "#profile-modal";
+const profileImgModal = "#edit-profile-image-modal";
 const profileName = ".profile__name";
 const profileDescription = ".profile__description";
+const profileImage = ".profile__image";
+const profileImageEl = document.querySelector(profileImage);
 const profileEditButton = document.querySelector(".profile__edit-button");
+const profileEditImageButton = document.querySelector(
+  ".profile__image__edit-button"
+);
 const addCardButton = document.querySelector(".profile__add-button");
 const addModal = "#add-modal";
+const editProfileImgForm = document.forms["edit-profile-image-form"];
 const addCardForm = document.forms["add-card-form"];
 const previewModal = "#preview-modal";
+const deleteModal = "#delete-modal";
 
-//add card section to DOM & add initial cards
+const api = new Api({
+  baseUrl: apiConfig.baseUrl,
+  headers: apiConfig.headers,
+});
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", imageClick);
-  return card.getView();
-}
+// Create user info instance
+const userInfo = new UserInfo({
+  userName: profileName,
+  userJob: profileDescription,
+});
+
+//receive user info
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      userName: res.name,
+      userJob: res.about,
+    });
+
+    profileImageEl.setAttribute("src", res.avatar);
+  })
+  .catch((err) => {
+    console.error("Failed to receive user info", err);
+  });
+
+//add card section to DOM & add initial cards
 
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (cardData) => {
+    renderer: (cardData, method) => {
+      if (cardData.link) {
+        cardData.cardLink = cardData.link;
+      }
+      if (cardData.name) {
+        cardData.cardName = cardData.name;
+      }
+
       const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement, "append");
+      cardSection.addItem(cardElement, method);
     },
   },
   ".cards__list"
 );
 
-cardSection.renderItems();
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards, "append");
+  })
+  .catch((err) => {
+    console.error("Failed to receive card data", err);
+  });
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    imageClick,
+    deleteClick,
+    likeClick
+  );
+  return card.getView();
+}
 
 //Submit handling functions
 
-function handleProfileFormSubmit(input) {
-  userInfo.setUserInfo(input);
-  profileEditPopup.close();
+function handleProfileFormSubmit(input, button) {
+  api
+    .setUserInfo(input)
+    .then((data) =>
+      userInfo.setUserInfo({ userName: data.name, userJob: data.about })
+    )
+    .catch((err) => {
+      console.error("Failed to save changes in user info", err);
+    })
+    .finally(() => {
+      profileEditPopup.close();
```

Comment:
` profileEditPopup.close();`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server)

**Please, correct it everywhere in the code**